lib/minitest/stub_const.rb

```diff
@@ -37,15 +36,21 @@ def stub_const(name, val, &block)
   # Example:
   #
   #   Object.stub_remove_const(:File) do
-  #     "Look ma, no File!" unless defined(File)
+  #     "Look ma, no File!" unless defined?(File)
   #   end
   #   # => "Look ma, no File!"
   def stub_remove_const(name)
-    orig = const_get(name)
-    remove_const(name)
-    yield
-  ensure
-    const_set(name, orig)
+    if const_defined?(name)
+      begin
+        orig = const_get(name)
+        remove_const(name)
+        yield
+      ensure
+        const_set(name, orig)
+      end
+    else
+      yield
```

Comment:
No, I think this makes sense. Thanks for elaborating. Would you mind expanding the docstring to clarify that the method does nothing (rather than e.g. raising) if the constant is undefined?